### DISCUSSION TITLE
How can I register my RegistrationToken on Azure after having asked permission and received it on my app?

### DISCUSSION THREAD

**Discussion by mwnovaprove** [shinyorg/shiny#1360] [Q&A] [answered]
I’m new to Shiny.NET and I’m trying to use it as a tool to implement push notifications on all OS (except Windows) while I need to use Azure Notifications Hub as a provider. I’m able to request permission from the OS and get a registration token, but then I don’t know how to register it on Azure. Would you please share with me some pieces of advice on how to do it? I tried hard with the documentation, but I read it’s not completed yet. 🙏🏽

**Comment by aritchie** [comment, marked as answer]
If you're using Shiny.Push.AzureNotificationHub, the registration token is for azure, as in, everything is already registered.

**Reply by mwnovaprove** [reply]
Thanks a lot for your answer! 🙏🏽 the only thing I don’t get is: why, trying to send a push notification through Azure with the “Test” function, I can’t receive it on Android emulator? Consider that I didn’t specify the token in the notification payload since, as for my knowledge of the functioning, I intended to send a broadcast notification.

**Reply by aritchie** [reply]
The test function as far as I can remember, sends to 10 random device registrations.  If you find an issue, please file it using our issue templates with all of the necessary details.

**Reply by mwnovaprove** [reply]
Thanks a lot @aritchie for your pieces of counsel! I'm able to get the notifications through the `OnReceived` method in the delegate. But I just wanted to ask you a couple of things:
- how do you suggest I can manage the local presentation of the incoming notifications? Is it by using Shiny Local Notifications? Because I see that unless I prompt someway the notification to the user, I can't see it neither on my notification bar nor anywhere else (in fact, on this account I can't properly understand when `OnEntry` in the delegate is fired). 
- For what is:
```
public async Task OnTokenChanged(string token)
{
    // fires when a push notification change is set by the operating system or provider
}
```
practically used? I'm asking this because I would like to have a clear idea of how everything works before implementing it on my application, since it's already in production. 
I thank you in advance for your kind reply!

**Reply by aritchie** [reply]
Notifications will show if you set the payloads properly.  I don't recommend anything here other than looking over the server push docs.
OnTokenChanged - is fired when the token changes outside of your general RequestAccess
